experimental/modular_oak_functions/src/demux.rs

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{Service, ServiceType};
+use std::{collections::BTreeMap, sync::Arc};
+
+/// A stream demultiplexer.
```

Comment:
Slightly confused by this, I read the earlier section on the design approach as indicating that a stream demultiplexer wouldn't be possible with the proposed design. 

> `ServiceProxy` instances that can be used to perform per-request service-related
> actions, and interact with the shared state owned by its parent service where
> needed.
> 
> Each service implementation would decide whether it must be able to call other
> services, and if so provide the ability to inject these services into a
> constructor. Each service will also be able to receive configuration information
> through a standardised interface.
> 
> This single approach would support most of the existing components in the Oak
> Functions runtime, but we will need some additional services for things that do
> not operate at the same per-request level. One example is a stream
> demultiplexer.